### ISSUE TITLE
Text wraps too late sometimes

### ISSUE THREAD

**Issue by jb55** [damus-io/notedeck#33]
Example of issue. Look at the center column in the before and after. In the center column the text on the right of the notes is cut off.

|before|after moving it a bit|
|------|---------------------|
<img src="https://github.com/emilk/egui/assets/45598/1b10f74a-478a-4073-b56d-93765f90ef5c" />|<img src="https://github.com/emilk/egui/assets/45598/8981d22d-5664-4ee4-a389-5f357bb91093"   />|

Apparently this was also an issue in gossip:

- https://github.com/emilk/egui/issues/2578 filed by @mikedilger

Fixed by @bu5hm4nn here: 

- https://github.com/emilk/egui/pull/2791

Let's try this branch and see if it fixes our issues.

**Comment by bu5hm4nn** [comment]
Yeah, there's a number of issues with horizontal_wrapped. We are currently running on "gossip-new-allpatch" branch on my fork https://github.com/bu5hm4nn/egui. You can cherry pick those patches onto 0.26 release. I already did a test with 0.27 but it needs a bit more work. (Also has some breaking changes again)

**Comment by jb55** [comment]
hmm ok we're using 0.27

**Comment by jb55** [comment]
updated to v0.29.1 and I think the issue got worse?

**Comment by kernelkind** [comment]
maybe it's a rounding issue: https://github.com/emilk/egui/issues/5163. In that case it would be fixed in https://github.com/emilk/egui/pull/5517, which hasn't made release yet. We could just wait for the next release and see if that fixes it

**Comment by kernelkind** [comment]
Need to try with https://github.com/emilk/egui/commit/dfcc679d5afaba3130356a2c6c922f7105f5a302 or later

**Comment by jb55** [comment]
ah

**Comment by mikedilger** [comment]
I have updated gossip to use egui 0.30 and in the process updated @bu5hm4nn's patches. 

The last 5 commits on this branch:   https://github.com/mikedilger/egui/tree/gossip_egui_0_30

I thought I remembered you trying these patches and them not working on notedeck, but I don't see that comment in this thread. Just disregard if this isn't helpful.

**Comment by jb55** [comment]
@mikedilger @bu5hm4nn all of my late wrapping issues was just because I had a ui.horizontal inside of a ui.horizontal_wrapped, which seemed to negate things and also break things after it

**Comment by mikedilger** [comment]
Interesting.

I've currently got an issue where if I have a form, and a button at the bottom right-aligned, and someone types a very long value into a text field in the form, the right-aligned button at the bottom slides rightward right off the damn window and you can't see it anymore without growing the window.
